File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = -1
    data.sort(key=lambda x: (x[col - 1], -x[0]))
    
    for row in range(row_begin, row_end + 1):    
        row_sum = sum(x % row for x in data[row - 1])
        target = int(bin(row_sum), 2)
        if answer == -1:
            answer = target
        else:
            answer ^= target
            
    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.Arrays;

class Solution {
    
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        Arrays.sort(data, (a, b) -> {
            if (a[col - 1] == b[col - 1]) {
                return b[0] - a[0];
            }
            return a[col - 1] - b[col - 1];
        });

        int answer = -1;

        for (int row = row_begin; row <= row_end; row++) {
            int rowSum = 0;
            for (int value : data[row - 1]) {
                rowSum += value % row;
            }

            int target = rowSum;
            answer = (answer == -1) ? target : answer ^ target;
        }

        return answer;
    }
}